### DISCUSSION TITLE
[vcpkg] Add command to print complete port description

### DISCUSSION THREAD

**Discussion by nurbles** [microsoft/vcpkg#17880] [Ideas]
I have searched the command line reference and FAQ and not found any way to have the `vcpkg search [pkg]` command to list the FULL text of the package description.  Many are quite short, but many others end with '...' after about 63 columns, regardless of how wide my terminal may be (using **Windows Terminal v 1.4.3243.0**)

If this option exists, what is it?

If not, please consider adding it, because it can be very helpful to read the full description of a package before installing it.

**Comment by JackBoosY** [comment]
I think we should add a `detail` option to vcpkg.

**Reply by dg0yt** [reply]
This would help even more for project manifest files, requested in https://github.com/microsoft/vcpkg/discussions/17938.

**Comment by JackBoosY** [comment]
And you can see the description in _VCPKG_ROOT/port/PORT_NAME/CONTROL_ or _VCPKG_ROOT/port/PORT_NAME/vcpkg.json_.

**Reply by dg0yt** [reply]
Now there is also the new package homepage: https://vcpkg.io/en/packages.html

**Comment by nurbles** [comment]
To clarify for those of us who only rarely use vcpkg, both of the places referenced where one may "see the description" are (or are not) available **WITHOUT** installing the package?

I am interested in this because my nearly 40 years of experience with all sorts of software packages (on both Windows and Linux) has shown that few completely and cleanly _remove_ themselves after having been installed.  That's why I **really** want to read the full description of a package _before_ I install it.

**Reply by dg0yt** [reply]
>  few completely and cleanly remove themselves after having been installed.

In vcpkg, the install command may do two things. Either it downloads the source, builds and installs, or it reuses a cached archive. In the second case, no specific scripts are run at install time, i.e. no side effects on existing files. In turn, port build commands cannot have effects other than installing new files. Removal should not be an issue from this point of view.

(This doesn't mean there couldn't be other pitfalls, such builds activating features based on previously installed ports which are not in the list of explicit dependencies.)

**Comment by NancyLi1013** [comment]
Related issue https://github.com/microsoft/vcpkg/issues/16286

**Comment by ras0219** [comment]
There is the option `--x-full-desc` which should do what you want:

```
$ vcpkg help search
The argument should be a substring to search for, or no argument to display all libraries.
Example:
  vcpkg search png

Options:
  --x-full-desc                   Do not truncate long text
...
```
